Classes/Bookmark/BookmarksViewController.swift

```diff
@@ -0,0 +1,220 @@
+        super.viewWillLayoutSubviews()
+    }
+    
+    // MARK: Private API
+    
+    func updateRightBarItem() {
+        navigationItem.rightBarButtonItem?.isEnabled = bookmarkStore.bookmarks.count > 0
+    }
+    
+    @objc private func onClearAll(sender: UIBarButtonItem) {
+        let alert = UIAlertController.configured(
+            title: NSLocalizedString("Are you sure?", comment: ""),
+            message: NSLocalizedString("All of your bookmarks will be lost. Do you want to continue?", comment: ""),
+            preferredStyle: .alert
+        )
+        
+        alert.addActions([
+            AlertAction.clearAll({ [weak self] _ in
+                self?.bookmarkStore.clear()
+                self?.tableView.reloadData()
+                self?.updateRightBarItem()
+            }),
+            AlertAction.cancel()
+        ])
+        
+        present(alert, animated: true)
+    }
+    
+    func filter(query: String?) {
+        if let query = query {
+            filterdBookmarks = filtered(array: bookmarkStore.bookmarks, query: query)
+        }
+        else {
+            filterdBookmarks = nil
+        }
+        self.tableView.reloadData()
+    }
+    
+    func getBookmarks() -> [BookmarkModel] {
+        if let bookmarks = filterdBookmarks {
+            return bookmarks
+        }
+        else {
+            return bookmarkStore.bookmarks
+        }
+    }
+    
+    // MARK: TabNavRootViewControllerType
+    
+    func didSingleTapTab() {
+        
+    }
+    
+    func didDoubleTapTab() {
+        // TODO: Scroll to top for table view
+    }
+    
+    // MARK: UITableViewDataSource
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return getBookmarks().count
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        var cell = tableView.dequeueReusableCell(withIdentifier: cellIdentifier)
+        
+        if cell == nil {
+            cell = UITableViewCell.init(style: .subtitle, reuseIdentifier: cellIdentifier)
+        }
+        
+        let bookmark = getBookmarks()[indexPath.row]
+        
+        let titleLabel = "\(bookmark.owner)/\(bookmark.name)"
+        cell?.textLabel?.text = bookmark.type == .repo ? titleLabel : titleLabel + " #\(bookmark.number)"
+        cell?.detailTextLabel?.text = bookmark.title
+        cell?.textLabel?.numberOfLines = 0
+        cell?.detailTextLabel?.numberOfLines = 0
+        cell?.accessibilityTraits |= UIAccessibilityTraitButton
+        cell?.isAccessibilityElement = true
+        cell?.accessibilityLabel = cell?.contentView.subviews
+            .flatMap { $0.accessibilityLabel }
+            .reduce("", { $0 + ".\n" + $1 })
+
+        cell?.imageView?.image = bookmark.type.icon?.withRenderingMode(.alwaysTemplate)
+        cell?.imageView?.tintColor = Styles.Colors.Blue.medium.color
+        return cell!
```

Comment:
Should we just create a subclass cell to incapsulate all this setup? Not a big deal since its only used here, but just an idea